data/transactions/logic/box.go

```diff
@@ -31,21 +32,37 @@ const (
 )
 
 func (cx *EvalContext) availableBox(name string, operation int, createSize uint64) error {
+	if cx.txn.Txn.OnCompletion == transactions.ClearStateOC {
+		return fmt.Errorf("boxes may not be accesses from ClearState program")
```

Comment:
@jannotti I added tests relating to the highlighted line via https://github.com/jannotti/go-algorand/pull/14.  Can you review to see if it's worth pulling in?

Reply:
merged.